core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -313,6 +313,46 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
             // E.g. "/usr/bin/docker run -v {{PROJECT_DIR}}:/project --rm quarkus/graalvm-native-image"
             nativeImage = new ArrayList<>();
             Collections.addAll(nativeImage, containerRuntime, "run", "-v", outputDir.toAbsolutePath() + ":/project:z", "--rm");
+            if (IS_LINUX & "docker".equals(containerRuntime)) {
+                try {
+                    BufferedReader reader;
+                    StringBuilder builder;
+                    String uid = null;
+                    String gid = null;
+
+                    ProcessBuilder idPB = new ProcessBuilder().command("id", "-ur");
+                    Process process = idPB.start();
+
+                    if (process.waitFor() == 0) {
+                        reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```

Comment:
I'm not familiar with this API, but is it safe to expect the input stream to contain the full output when the process is already terminated?

Reply:
No, you normally should consume the output first.  But in this case since Linux uses (IIRC) a 4k buffer and we have only a couple bytes to read, it coincidentally works.

The error output should also be consumed or (better yet) redirected to `/dev/null` (Java 11 or 12 has a constant for this but for us the literal should be fine for now since native image is presently non-Windows).  The process output stream (i.e. `stdin`) should be immediately closed.

Reply:
I have refactored. I am now redirecting error stream to `/dev/null`. This is only for Linux builds, as docker runs the container as root by default on docker.   I am closing the output stream as soon as the process is started, and I am consuming the input stream in a loop and not waiting for the process to terminate.  It was working before as the data returned was only a few bytes, so it was available in the buffer.  I am also closing the error stream, input stream and the buffered reader after the process terminates. I will re-push